src/libstd/net/tcp.rs

```diff
@@ -498,18 +498,45 @@ impl TcpStream {
 
     /// Moves this TCP stream into or out of nonblocking mode.
     ///
-    /// On Unix this corresponds to calling fcntl, and on Windows this
-    /// corresponds to calling ioctlsocket.
+    /// This will result in `read`, `write`, `recv` and `send` operations
+    /// becoming nonblocking, i.e. immediately returning from their calls.
+    /// If the IO operation is successful, `Ok` is returned and no further
+    /// action is required. If the IO operation is pending, an error with kind
+    /// [`io::ErrorKind::WouldBlock`] is returned.
+    ///
+    /// On Unix platforms, calling this method corresponds to calling `fcntl`.
+    /// On Windows calling this method corresponds to calling `ioctlsocket`.
```

Comment:
It is somewhat import to specify which `fcntl` or `ioctlsocket` is used, both `fcntl` and `ioctlsocket` are some kind of "catch-all" syscall that can do wildly differing things. The `fcntl` `FIONBIO` is used on Unix platforms and the `ioctlsocket` `FIONBIO` is used on Windows.

Reply:
addressed in https://github.com/rust-lang/rust/pull/45227/commits/c978e7317142d9cea7c88b329924438116516fdc